XIPs/xip-55-passkey-identity.md

```diff
@@ -0,0 +1,166 @@
+
+#### Phase 2
+
+Phase 2 will add a Solana `MemberIdentifier`, and will *not* break compatibility with Phase 1 or newer LibXMTP releases due to new `MemberIdentifiers` not being able to sign identity updates, and the graceful handling of new unknown `MemberIdentifiers`.
+
+### Inbox state
+
+Inbox state has both a `recovery_address: String` field and an `addresses: Vec<String>` field. Because XMTP will now be supporting more chains than just Ethereum, this needs to change. First, to draw attention to the change, the fields will be renamed to `recovery_identifier` and `identifiers`, and the `String` type will be changed to a new struct: `PublicIdentifier` which is basically a more public-facing version of `MemberIdentifier`. The updated `InboxState` will look like this:
+
+```rust
+#[napi(object)]
+pub struct InboxState { 
+  pub inbox_id: String,
+  pub recovery_identifier: PublicIdentifier,
+  pub installations: Vec<Installation>,
+  pub identifiers: Vec<PublicIdentifier>,
+}
+
+#[napi(object)]
+pub enum PublicIdentifier {
+  Ethereum(String),
+  Passkey(Vec<u8>),
+}
+
+impl PublicIdentifier {
+  fn display(&self) -> String {
+    // ...
+  }
+}
+```
+
+## Reference implementation
+
+### High-level technical implementation
+
+#### Phase 1: Adding passkey support (breaking)
+
+- Add a new `MemberIdentifier::Passkey` variant
+  - The `Passkey` variant needs to record the relying party identifier along with the public key.
+  - Rename `MemberIdentifier::Address` to `MemberIdentifier::EthAddress` to distinguish from the new `MemberIdentifier` variants coming in
+- Add a new `UnverifiedSignature::Passkey` variant
+  - Implement the associated `to_verified` branch logic
+- Add a new `SignatureKind::P256` variant
+- Rename `associate_wallet` to `associate_member` and change it to take in a `MemberIdentifier` enum instead of a `new_wallet_address`, and block on the `MemberIdentifier::Installation` variant.
+- Adapt [generate_inbox_id](https://github.com/xmtp/libxmtp/blob/71b47a20020ecece139cdc0d56ab356d3f6637f1/xmtp_id/src/associations/hashes.rs#L22) to work with passkey addresses, mainly updating the valid address check.
+- Create an `Identifier` type, which is a public facing version of `MemberIdentifier` and doesn’t contain potentially sensitive metadata like the passkey’s relying partner.
+- Rename `InboxState.account_addresses` to `InboxState.identifiers` and change the type to `Vec<Identifier>`
+  - Also rename `InboxState.recovery_address` to `recovery_identifier` and change the type to `Identifier`
+
+#### Phase 2: Adding Solana support (non-breaking)
+
+- Add a new `MemberIdentifier::Solana` variant
+- XMTP will **not** be adding Solana variants for `UnverifiedSignature` or `SignatureKind`
+
+This means that the `MemberIdentifier` in the `AddAssociation` proto may come through as `None` in old versions that haven't received the Solana update. It must be verified that this behavior is acceptable.
```

Comment:
We should mention this in Phase 1 as well, since protection for this will need to be implemented there. In phase 1 can we add

```
- Filter out unknown MemberIdentifiers that may come from new associations.
```